distributed/client.py

```diff
@@ -3642,46 +3650,43 @@ def restart(self, timeout=no_default, wait_for_workers=True):
     async def _restart_workers(
         self,
         workers: list[str],
-        timeout: int | float | None = None,
-        raise_for_error: bool = True,
-    ) -> dict[str, str | ErrorMessage]:
+        timeout: str | int | float | NoDefault,
+        raise_for_error: bool,
+    ) -> dict[str, Literal["OK", "timed out"]]:
```

Comment:
```suggestion
    ) -> dict[str, Literal["OK", "removed", "timed out"]]:
```
IIUC, we return `"removed"` for non-nanny workers.